src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/type/model/AbstractTransactionModel.java

```diff
@@ -2,21 +2,25 @@
 
 import com.kryptokrauts.aeternity.generated.api.rxjava.ExternalApi;
 import com.kryptokrauts.aeternity.generated.model.GenericTx;
+import com.kryptokrauts.aeternity.sdk.annotations.Mandatory;
 import com.kryptokrauts.aeternity.sdk.service.transaction.type.AbstractTransaction;
 import com.kryptokrauts.sophia.compiler.generated.api.rxjava.DefaultApi;
+import java.lang.reflect.Field;
 import java.math.BigInteger;
 import java.util.function.Function;
 import lombok.Getter;
 import lombok.Setter;
 import lombok.experimental.SuperBuilder;
+import lombok.extern.slf4j.Slf4j;
 
 @Getter
 @Setter
 @SuperBuilder(toBuilder = true)
+@Slf4j
 public abstract class AbstractTransactionModel<GeneratedTxModel> {
 
   /** the fee is automatically calculated but can also be set manually */
-  protected BigInteger fee;
+  @Mandatory protected BigInteger fee;
```

Comment:
@mitch-lbw should we remove this annotation here? reasons:
- the field isn't checked and the annotation can therefore can be misleading
- the fee is always calculated so the check isn't really needed